File: archive/mobile-app-ios-legacy/Plagit/Service/ServiceSubscriptionView.swift
//
//  ServiceSubscriptionView.swift
//  Plagit
//
//  Service provider subscription paywall — 3 tiers: Basic, Pro, Premium.
//

import SwiftUI
import StoreKit

struct ServiceSubscriptionView: View {
    @Environment(\.dismiss) private var dismiss
    @State private var sub = SubscriptionManager.shared
    @State private var selectedTier: Tier = .pro
    @State private var isAnnual = true

    enum Tier: String, CaseIterable { case basic, pro, premium }

    private struct TierInfo {
        let name: String
        let price: String
        let annualPrice: String
        let features: [(icon: String, text: String)]
        let color: Color
        let isPopular: Bool
    }

    private func tierInfo(_ tier: Tier) -> TierInfo {
        switch tier {
        case .basic:
            return TierInfo(name: "Basic", price: "$9.99/mo", annualPrice: "$99.99/yr", features: [
                ("checkmark.seal", "Verified Badge"),
                ("arrow.up.circle", "Improved Visibility"),
                ("doc.text", "Up to 10 Posts"),
                ("person.2", "20 Contacts/Month"),
            ], color: .plagitTeal, isPopular: false)
        case .pro:
            return TierInfo(name: "Pro", price: "$19.99/mo", annualPrice: "$199.99/yr", features: [
                ("sparkles", "Featured Placement"),
                ("chart.bar", "Advanced Analytics"),
                ("doc.text.fill", "Up to 50 Posts"),
                ("person.2.fill", "100 Contacts/Month"),
                ("list.star", "Priority Listing"),
            ], color: .plagitAmber, isPopular: true)
        case .premium:
            return TierInfo(name: "Premium", price: "$39.99/mo", annualPrice: "$399.99/yr", features: [
                ("crown", "Premium Badge"),
                ("arrow.up.circle.fill", "Top Placement"),
                ("bolt.fill", "Feed Boost"),
                ("chart.bar.fill", "Full Analytics"),
                ("infinity", "Unlimited Posts"),
                ("infinity", "Unlimited Contacts"),
            ], color: .plagitIndigo, isPopular: false)
        }
    }

    var body: some View {
        ZStack {
            Color.plagitBackground.ignoresSafeArea()
            ScrollView(.vertical, showsIndicators: false) {
                VStack(spacing: PlagitSpacing.lg) {
                    topBar

                    // Header
                    VStack(spacing: PlagitSpacing.sm) {
                        Text("Grow Your Service Business")
                            .font(.system(size: 22, weight: .bold, design: .rounded))
                            .foregroundColor(.plagitCharcoal)
                        Text("Get discovered by more clients with a premium listing.")
                            .font(PlagitFont.body())
                            .foregroundColor(.plagitSecondary)
                            .multilineTextAlignment(.center)
                    }
                    .padding(.horizontal, PlagitSpacing.xl)

                    // Annual toggle
                    HStack(spacing: PlagitSpacing.md) {
                        Text("Monthly").font(PlagitFont.captionMedium()).foregroundColor(isAnnual ? .plagitTertiary : .plagitCharcoal)
                        Toggle("", isOn: $isAnnual).labelsHidden().tint(.plagitAmber)
                        HStack(spacing: PlagitSpacing.xs) {
                            Text("Annual").font(PlagitFont.captionMedium()).foregroundColor(isAnnual ? .plagitCharcoal : .plagitTertiary)
                            Text("Save 17%").font(PlagitFont.micro()).foregroundColor(.plagitOnline)
                                .padding(.horizontal, 6).padding(.vertical, 2)
                                .background(Capsule().fill(Color.plagitOnline.opacity(0.1)))
                        }
                    }

                    // Tier cards
                    ForEach(Tier.allCases, id: \.self) { tier in
                        tierCard(tier)
                    }

                    // Current plan info
                    if sub.currentPlan.isServiceAny {
                        HStack(spacing: PlagitSpacing.sm) {
                            Image(systemName: "checkmark.circle.fill").foregroundColor(.plagitOnline)
                            Text("Current plan: \(sub.currentPlan.displayName)")
                                .font(PlagitFont.captionMedium()).foregroundColor(.plagitCharcoal)
                        }
                        .padding(PlagitSpacing.lg)
                        .frame(maxWidth: .infinity)
                        .background(RoundedRectangle(cornerRadius: PlagitRadius.md).fill(Color.plagitOnline.opacity(0.06)))
                        .padding(.horizontal, PlagitSpacing.xl)
                    }

                    // Subscribe button
                    Button {
                        // TODO: purchase service tier via StoreKit
                    } label: {
                        Text("Subscribe to \(tierInfo(selectedTier).name)")
                            .font(PlagitFont.subheadline())
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, PlagitSpacing.lg)
                            .background(
                                Capsule().fill(
                                    LinearGradient(colors: [tierInfo(selectedTier).color, tierInfo(selectedTier).color.opacity(0.8)], startPoint: .leading, endPoint: .trailing)
                                )
                            )
                    }
                    .padding(.horizontal, PlagitSpacing.xl)

                    // Restore
                    Button { Task { await sub.restorePurchases() } } label: {
                        Text("Restore Purchases").font(PlagitFont.caption()).foregroundColor(.plagitTertiary)
                    }

                    Spacer().frame(height: PlagitSpacing.xxxl)
                }
            }
        }
        .navigationBarHidden(true)
    }

    // MARK: - Top Bar

    private var topBar: some View {
        HStack {
            Button { dismiss() } label: {
                Image(systemName: "chevron.left").font(.system(size: 18, weight: .medium))
                    .foregroundColor(.plagitCharcoal).frame(width: 36, height: 36)
            }
            Spacer()
            Text("Service Plans").font(PlagitFont.subheadline()).foregroundColor(.plagitCharcoal)
            Spacer()
            Color.clear.frame(width: 36, height: 36)
        }
        .padding(.horizontal, PlagitSpacing.xl).padding(.top, PlagitSpacing.lg).padding(.bottom, PlagitSpacing.sm)
    }

    // MARK: - Tier Card

    private func tierCard(_ tier: Tier) -> some View {
        let info = tierInfo(tier)
        let isSelected = selectedTier == tier

        return Button { withAnimation(.easeInOut(duration: 0.2)) { selectedTier = tier } } label: {
            VStack(alignment: .leading, spacing: PlagitSpacing.md) {
                HStack {
                    VStack(alignment: .leading, spacing: 2) {
                        HStack(spacing: PlagitSpacing.sm) {
                            Text(info.name).font(PlagitFont.headline()).foregroundColor(.plagitCharcoal)
                            if info.isPopular {
                                Text("Most Popular").font(PlagitFont.micro()).foregroundColor(.white)
                                    .padding(.horizontal, 6).padding(.vertical, 2)
                                    .background(Capsule().fill(info.color))
                            }
                        }
                        Text(isAnnual ? info.annualPrice : info.price)
                            .font(PlagitFont.bodyMedium()).foregroundColor(info.color)
                    }
                    Spacer()
                    Image(systemName: isSelected ? "checkmark.circle.fill" : "circle")
                        .font(.system(size: 22)).foregroundColor(isSelected ? info.color : .plagitBorder)
                }

                ForEach(info.features, id: \.text) { feature in
                    HStack(spacing: PlagitSpacing.sm) {
                        Image(systemName: feature.icon).font(.system(size: 12, weight: .medium))
                            .foregroundColor(info.color).frame(width: 18)
                        Text(feature.text).font(PlagitFont.caption()).foregroundColor(.plagitCharcoal)
                    }
                }
            }
            .padding(PlagitSpacing.xl)
            .background(
                RoundedRectangle(cornerRadius: PlagitRadius.xl)
                    .fill(Color.plagitCardBackground)
                    .overlay(
                        RoundedRectangle(cornerRadius: PlagitRadius.xl)
                            .stroke(isSelected ? info.color : Color.plagitBorder.opacity(0.3), lineWidth: isSelected ? 2 : 1)
                    )
                    .shadow(color: isSelected ? info.color.opacity(0.1) : .clear, radius: 8, y: 4)
            )
        }
        .buttonStyle(.plain)
        .padding(.horizontal, PlagitSpacing.xl)
    }
}

File: mobile-app-flutter/lib/features/services/views/service_subscription_view.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:plagit/config/app_theme.dart';

/// Service provider subscription paywall -- 3 tiers: Basic, Pro, Premium.
/// Mirrors ServiceSubscriptionView.swift.
class ServiceSubscriptionView extends StatefulWidget {
  const ServiceSubscriptionView({super.key});

  @override
  State<ServiceSubscriptionView> createState() => _ServiceSubscriptionViewState();
}

enum _Tier { basic, pro, premium }

class _TierInfo {
  final String name;
  final String price;
  final String annualPrice;
  final List<(IconData, String)> features;
  final Color color;
  final bool isPopular;

  const _TierInfo({
    required this.name,
    required this.price,
    required this.annualPrice,
    required this.features,
    required this.color,
    this.isPopular = false,
  });
}

final Map<_Tier, _TierInfo> _tierData = {
  _Tier.basic: const _TierInfo(
    name: 'Basic',
    price: '\$9.99/mo',
    annualPrice: '\$99.99/yr',
    features: [
      (Icons.verified, 'Verified Badge'),
      (Icons.trending_up, 'Improved Visibility'),
      (Icons.article, 'Up to 10 Posts'),
      (Icons.people, '20 Contacts/Month'),
    ],
    color: AppColors.teal,
  ),
  _Tier.pro: const _TierInfo(
    name: 'Pro',
    price: '\$19.99/mo',
    annualPrice: '\$199.99/yr',
    features: [
      (Icons.auto_awesome, 'Featured Placement'),
      (Icons.bar_chart, 'Advanced Analytics'),
      (Icons.article_outlined, 'Up to 50 Posts'),
      (Icons.people_alt, '100 Contacts/Month'),
      (Icons.star, 'Priority Listing'),
    ],
    color: AppColors.amber,
    isPopular: true,
  ),
  _Tier.premium: const _TierInfo(
    name: 'Premium',
    price: '\$39.99/mo',
    annualPrice: '\$399.99/yr',
    features: [
      (Icons.workspace_premium, 'Premium Badge'),
      (Icons.arrow_circle_up, 'Top Placement'),
      (Icons.bolt, 'Feed Boost'),
      (Icons.bar_chart_outlined, 'Full Analytics'),
      (Icons.all_inclusive, 'Unlimited Posts'),
      (Icons.all_inclusive, 'Unlimited Contacts'),
    ],
    color: AppColors.indigo,
  ),
};

class _ServiceSubscriptionViewState extends State<ServiceSubscriptionView> {
  _Tier _selectedTier = _Tier.pro;
  bool _isAnnual = true;

  @override
  Widget build(BuildContext context) {
    final selected = _tierData[_selectedTier]!;

    return Scaffold(
      backgroundColor: AppColors.background,
      body: SafeArea(
        child: SingleChildScrollView(
          child: Column(
            children: [
              // Top bar
              Padding(
                padding: const EdgeInsets.fromLTRB(AppSpacing.xl, AppSpacing.lg, AppSpacing.xl, AppSpacing.sm),
                child: Row(
                  children: [
                    GestureDetector(
                      onTap: () => context.pop(),
                      child: const SizedBox(width: 36, height: 36, child: Icon(Icons.chevron_left, size: 22, color: AppColors.charcoal)),
                    ),
                    const Spacer(),
                    const Text('Service Plans', style: TextStyle(fontSize: 15, fontWeight: FontWeight.w500, color: AppColors.charcoal)),
                    const Spacer(),
                    const SizedBox(width: 36, height: 36),
                  ],
                ),
              ),

              // Header
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: AppSpacing.xl),
                child: Column(
                  children: [
                    const SizedBox(height: AppSpacing.sm),
                    const Text(
                      'Grow Your Service Business',
                      style: TextStyle(fontSize: 22, fontWeight: FontWeight.bold, color: AppColors.charcoal),
                    ),
                    const SizedBox(height: AppSpacing.sm),
                    const Text(
                      'Get discovered by more clients with a premium listing.',
                      textAlign: TextAlign.center,
                      style: TextStyle(fontSize: 15, color: AppColors.secondary),
                    ),
                  ],
                ),
              ),
              const SizedBox(height: AppSpacing.lg),

              // Annual toggle
              Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Text('Monthly', style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: _isAnnual ? AppColors.tertiary : AppColors.charcoal)),
                  const SizedBox(width: AppSpacing.md),
                  Switch(
                    value: _isAnnual,
                    activeColor: AppColors.amber,
                    onChanged: (v) => setState(() => _isAnnual = v),
                  ),
                  const SizedBox(width: AppSpacing.xs),
                  Text('Annual', style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: _isAnnual ? AppColors.charcoal : AppColors.tertiary)),
                  const SizedBox(width: AppSpacing.xs),
                  Container(
                    padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                    decoration: BoxDecoration(color: AppColors.online.withValues(alpha: 0.1), borderRadius: BorderRadius.circular(AppRadius.full)),
                    child: const Text('Save 17%', style: TextStyle(fontSize: 11, color: AppColors.online)),
                  ),
                ],
              ),
              const SizedBox(height: AppSpacing.lg),

              // Tier cards
              ..._Tier.values.map((tier) => Padding(
                padding: const EdgeInsets.only(bottom: AppSpacing.md),
                child: _tierCard(tier),
              )),

              // Subscribe button
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: AppSpacing.xl),
                child: GestureDetector(
                  onTap: () {},
                  child: Container(
                    width: double.infinity,
                    padding: const EdgeInsets.symmetric(vertical: AppSpacing.lg),
                    decoration: BoxDecoration(
                      gradient: LinearGradient(colors: [selected.color, selected.color.withValues(alpha: 0.8)]),
                      borderRadius: BorderRadius.circular(AppRadius.full),
                    ),
                    alignment: Alignment.center,
                    child: Text(
                      'Subscribe to ${selected.name}',
                      style: const TextStyle(fontSize: 15, fontWeight: FontWeight.w600, color: Colors.white),
                    ),
                  ),
                ),
              ),
              const SizedBox(height: AppSpacing.md),

              // Restore purchases
              GestureDetector(
                onTap: () {},
                child: const Text('Restore Purchases', style: TextStyle(fontSize: 13, color: AppColors.tertiary)),
              ),
              const SizedBox(height: AppSpacing.xxxl),
            ],
          ),
        ),
      ),
    );
  }

  Widget _tierCard(_Tier tier) {
    final info = _tierData[tier]!;
    final isSelected = _selectedTier == tier;

    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: AppSpacing.xl),
      child: GestureDetector(
        onTap: () => setState(() => _selectedTier = tier),
        child: AnimatedContainer(
          duration: const Duration(milliseconds: 200),
          padding: const EdgeInsets.all(AppSpacing.xl),
          decoration: BoxDecoration(
            color: AppColors.cardBackground,
            borderRadius: BorderRadius.circular(AppRadius.xl),
            border: Border.all(
              color: isSelected ? info.color : AppColors.border.withValues(alpha: 0.3),
              width: isSelected ? 2 : 1,
            ),
            boxShadow: isSelected
                ? [BoxShadow(color: info.color.withValues(alpha: 0.1), blurRadius: 8, offset: const Offset(0, 4))]
                : null,
          ),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Row(
                children: [
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: [
                            Text(info.name, style: const TextStyle(fontSize: 18, fontWeight: FontWeight.w600, color: AppColors.charcoal)),
                            if (info.isPopular) ...[
                              const SizedBox(width: AppSpacing.sm),
                              Container(
                                padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                                decoration: BoxDecoration(color: info.color, borderRadius: BorderRadius.circular(AppRadius.full)),
                                child: const Text('Most Popular', style: TextStyle(fontSize: 11, color: Colors.white)),
                              ),
                            ],
                          ],
                        ),
                        const SizedBox(height: 2),
                        Text(
                          _isAnnual ? info.annualPrice : info.price,
                          style: TextStyle(fontSize: 15, fontWeight: FontWeight.w500, color: info.color),
                        ),
                      ],
                    ),
                  ),
                  Icon(
                    isSelected ? Icons.check_circle : Icons.circle_outlined,
                    size: 24,
                    color: isSelected ? info.color : AppColors.border,
                  ),
                ],
              ),
              const SizedBox(height: AppSpacing.md),
              ...info.features.map((f) => Padding(
                padding: const EdgeInsets.only(bottom: AppSpacing.sm),
                child: Row(
                  children: [
                    Icon(f.$1, size: 14, color: info.color),
                    const SizedBox(width: AppSpacing.sm),
                    Text(f.$2, style: const TextStyle(fontSize: 13, color: AppColors.charcoal)),
                  ],
                ),
              )),
            ],
          ),
        ),
      ),
    );
  }
}
